### DISCUSSION TITLE
Communicating repo status "importance" / "desire for contributions"?

### DISCUSSION THREAD

**Discussion by justinabrahms** [todogroup/ospology#149] [Community Proposals]
In a recent touchpoint, we talked about the investment put into a given project. We have https://opensource.twitter.dev/status/ which tracks repo stability, but we don't have a way to talk about how important the repo is in to the OSPO.

What ways would we talk about this?

My guess:
- "source code dump"
- "we think it would be cool if you used this"
- "it's actually a 'big bet' for the company and we want to facilitate a community around this"

**Comment by detiber** [comment]
I think @juliaferraioli captured this a bit in her blog around purpose of open source here: https://www.juliaferraioli.com/blog/2022/social-model-oss/#breaking-down-open-source-projects-by-purpose

**Comment by juliaferraioli** [comment]
There are two aspects here: internal and external

For use *inside* a company when talking about projects yet to be released as open source, I generally categorize things as (in no particular order):

- hobby project
- small project
- small strategic project
- experimental project
- large strategic project

For use *outside* the company context when telegraphing how the company thinks about the project, these are some categories that i have used (in no particular order and *not* mutually exclusive):

- hobby project
- experimental / incubating
- demo / example
- static / proof of concept
- used in production
- others I can't think of

Generally, I'd put lots of this information in matrix form:

| name | description | type | status | governance approach | usage | accepting contributions | other |
| --- | --- | --- | --- | --- | --- | --- | --- |
| super awesome project | this is going to change the cloud native game forever | large strategic project | experimental | uh, we should have a plan | used at company X | yes | let's build a brand around this, please help |

Or some such. Whatever matrix you use can help you categorize and communicate both internally and externally.

**Comment by jeffpaul** [comment]
We've codified this a bit @10up via https://10up.github.io/Open-Source-Best-Practices/github-process/#support-levels (noting this was first identified and replicated from https://formidable.com/blog/2019/oss-maintenance/).

**Comment by nicorikken** [comment]
The French government uses 4 levels: https://guides.etalab.gouv.fr/logiciels/#a-quoi-sert-il

- 📘 **Niveau A - contributif** : _Le code source est publié, les contributions extérieures sont activement recherchées et traitées._: open source and actively looking for external contributions 
- 📗 **Niveau B - ouvert** : _Le code source est publié, les contributions extérieures sont traitées mais non activement recherchées._: open source, external contributions will be considered/handled but won't be sought after
-  📙 **Niveau C - publié**: _Le code source est publié mais les contributions extérieures ne sont pas traitées._ : open source but offered contributions won't be considered.
- 📕 **Niveau D - non-communicable**: _Le code source n'est pas communicable au public._ : source code can't be published

This is mainly focussed on external contributions, but I think that is most relevant to communicate. The way I see it the level can change after a couple of years if the situation changes.
